docs/advanced-topics/advanced-losses-optimizers-and-poolers.md

```diff
@@ -0,0 +1,102 @@
+
+SphereFace loss is a loss function that was first formulated for computer vision and face recognition tasks.
+Finetuner supports two variations of this loss function, `ArcFaceLoss` and `CosFaceLoss`.
+Instead of attempting to minimise the distance between positive pairs and maximise the distance between negative pairs, the SphereFace loss functions compare each sample with the center of each class
+and attempt to minimize the *angular distance* between the document and its class centroid, and maximise the angular distance between the document and the centroids of the other classes.
+
+![training](../imgs/SphereFace-training.png)
+
+The `ArcFaceLoss` and `CosFaceLoss` both deviate from the traditional SphereFace loss by including a margin and scaling parameter, which can be used to increase the boundary between each class. If an item's embedding is within the boundary of the class it belongs to, then no loss is incurred. Choosing appropriate values for the margin and scaling parameter is important for effective training, for more information on how `ArcFaceLoss` and `CosFaceLoss` calculate loss, and how these parameters affect the output, see this article on [loss metrics for deep learning](https://hav4ik.github.io/articles/deep-metric-learning-survey#cosface).  
+
+`TripletMarginLoss` uses a `ClassSampler` to construct batches with an equal number of samples of each class in the batch. However, since only one sample is needed to calculate the loss with the `ArcFaceLoss` and `CosFaceLoss` functions, there are no constraints on what each batch needs to contain.
+Therefore we can instead construct batches using random sampling, a much simpler method which consequently takes less time to construct a batch.
+By default, runs created using `ArcFaceLoss` or `CosfaceLoss` will use random sampling, however you can specify which type of sampling method you would like to use like so:
+
+```diff
+run = finetuner.fit(
+    ...,
+    loss='ArcFaceLoss',
++   sampler='random'      # use random sampling
++   sampler='class'       # use class sampling
++   sampler='auto'        # infer sampling method based on the loss function (default)
+
+)
+```
+
+Since `TripletMarginLoss` needs a class sampler to properly function, this `sampler` parameter is ignored when `loss='TripletMarginLoss'`.
+
+### Using an Optimizer
+
+In order to keep track of the class centers across batches, these SphereFace loss functions require an additional optimizer during training.
+By default, the type of optimizer used will be the same as the one used for the model itself, but you can also choose a different optimizer for your loss function using the `loss_optimizer` parameter.
+The list of available optimizers are discussed in the [run job](../walkthrough/run-job.md) section.
+
+```diff
+run = finetuner.fit(
+    ...,
+    loss='ArcFaceLoss',
++   loss_optimizer='Adam',
++   loss_optimizer_options={'weight_decay': 0.01}
+)
+```
+
+### Comparing with `TripletMarginLoss`
```

Comment:
maybe remove the quotes in the headline. It looks a bit strange, and you also don't do it consistently, but only here